cli/cmd/inject.go

```diff
@@ -407,12 +458,22 @@ func injectResource(bytes []byte, options *injectOptions) ([]byte, error) {
 
 	}
 
+	// The namespace isn't necessarily in the input so it has to be substituted
+	// at runtime. The proxy recognizes the "$NAME" syntax for this variable
+	// but not necessarily other variables.
+	identity := k8s.TLSIdentity {
+		Name: name,
```

Comment:
There's no need for setting a separate `name` variable in each of the cases above -- we're already capturing the object metadata for each object type in `objectMetada`, and that includes the object's name. I'd revert the changes from lines 364-488, and instead write this as:

```
	name := ""
	if objectMeta != nil {
		name = objectMeta.Name
	}

	identity := k8s.TLSIdentity{
		Name:                name,
		Kind:                k8s.GetOwnerTypeFromLabels(k8sLabels),
		Namespace:           "$" + PodNamespaceEnvVarName,
		ControllerNamespace: controlPlaneNamespace,
	}
```

---

Note: you need the `objectMeta != nil` check because we might be dealing with an object that's not covered by any of the listed cases above, such as a namespace. You could avoid the `objectMeta != nil` check though if you add a `default` case to the switch statement:

```
	default:
		return bytes, nil
	}

	// The namespace isn't necessarily in the input so it has to be substituted
	// at runtime. The proxy recognizes the "$NAME" syntax for this variable
	// but not necessarily other variables.
	identity := k8s.TLSIdentity{
		Name:                objectMeta.Name,
		Kind:                k8s.GetOwnerTypeFromLabels(k8sLabels),
		Namespace:           "$" + PodNamespaceEnvVarName,
		ControllerNamespace: controlPlaneNamespace,
	}
```

I like that approach better since we can return immediately once we've determined that we're not dealing with an object that we can inject. Not having a `default` case on the switch statement seems like an oversight.

Reply:
I did it more like your first suggestion instead of adding the default case, because the code was already conditionally returning `bytes` unmodified.